plugins/destination/s3/client/write.go

```diff
@@ -50,3 +64,24 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data
 
 	return nil
 }
+
+// sanitizeJSONKeys replaces all invalid characters in JSON keys with underscores.
+// It does the replacement in-place, modifying the original object. This is required
+// for compatibility with Athena.
+func sanitizeJSONKeys(obj any) {
+	switch m := obj.(type) {
+	case map[string]any:
```

Comment:
This wouldn't match `map[string]string` etc, if we have those kinds of objects coming through here. Could be edge cases.

Reply:
Hmm true, let me add some cases for this!